### PR TITLE
Remove K, V from StateChange in State and Receipts module

### PR DESCRIPTION
**Pull request by shannynalayna**
Removes the K & V generic types from the StateChange object in the State and Receipts module. This change also affects the TransactionReceipt object. Changes were also necessary for the implementation of the ContextManager to deal with the String and Vec<u8> type directly. 
This pr also makes changes in the State module as the HashMapState was previously implemented with <String, String> for <K, V>.

### CODE REVIEW

File: libtransact/src/context/manager.rs [hunk 1]
```diff
@@ -173,8 +162,8 @@ impl<
     pub fn set_state(
         &mut self,
         context_id: &ContextId,
-        key: K,
-        value: V,
+        key: &str,
```
Comment (peterschwarz) [inline libtransact/src/context/manager.rs]: These should be owned `String` and `Vec<u8>`, as the previous types were owned.

File: libtransact/src/context/manager.rs [hunk 2]
```diff
@@ -119,11 +112,12 @@ impl<
 
     /// Get the values associated with list of keys, from a specific Context.
     /// If a key is not found in the context, State is then checked for these keys.
+    /// Keys are returned with the associated value, if found in Context or State.
     pub fn get(
         &self,
         context_id: &ContextId,
-        keys: &[K],
-    ) -> Result<Vec<(K, Option<V>)>, ContextManagerError> {
+        keys: &[&str],
```
Comment (peterschwarz) [inline libtransact/src/context/manager.rs]: this should probably be `&[String]` as converting a `Vec` of `String` to a slice of `&str` is quite difficult.

File: libtransact/src/context/mod.rs [hunk 3]
```diff
@@ -98,35 +97,42 @@ impl<K: PartialEq + Clone + Debug, V: Clone + Debug> Context<K, V> {
     }
 
     /// Adds StateChange::Set without deleting previous StateChanges associated with the Key
-    pub fn set_state(&mut self, key: K, value: V) {
-        let new_state_change = StateChange::Set { key, value };
+    pub fn set_state(&mut self, key: &str, value: &[u8]) {
```
Comment (peterschwarz) [inline libtransact/src/context/mod.rs]: These should be owned values as the previous signature.

File: libtransact/src/context/manager.rs (hunk 1 above)
Comment (shannynalayna) [reply libtransact/src/context/manager.rs]: Addressed

File: libtransact/src/context/manager.rs (hunk 2 above)
Comment (shannynalayna) [reply libtransact/src/context/manager.rs]: Addressed

File: libtransact/src/context/mod.rs (hunk 3 above)
Comment (shannynalayna) [reply libtransact/src/context/mod.rs]: Addressed

**Pull request merged**